### ISSUE TITLE
HOLD [$250] Account not showing recent expenses/scans in my workspace chat

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#45546]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:**
**Reproducible in staging?:** n/a
**Reproducible in production?:** n/a
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @danielrvidal 
**Slack conversation:** https://expensify.slack.com/archives/C05LX9D6E07/p1721152209600379

## Action Performed:
1. Scan a receipt and completed successfully
2. Expense from yesterday is missing
3. Observe all the receipts scanned previously
## Expected Result:
All the past expense should be available
## Actual Result:

- All expenses just kept saying “Receipt scanning…” even though it was showing expenses from a month ago.
- In the video, you can see me switch to New Expensify the app rather than hybrid app and it shows my expenses.
- Seems some expenses stuck in a scanning state. After a while of having the app open, expense from yesterday came over and the one from today scanned successfully. 
- But the two from June 17th remain in a “Receipt Scanning…” state.

## Workaround:
unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [x] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos
[onyx-state.txt](https://github.com/user-attachments/files/16257923/onyx-state.txt)

[logs-2024-07-16 17_45_38.974.txt](https://github.com/user-attachments/files/16257924/logs-2024-07-16.17_45_38.974.txt)

https://github.com/user-attachments/assets/f9ebfc87-8b27-4e4b-ab1a-15d401c09442

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0184e98d80e32fd2e3</li>
        <li>Upwork Job ID: 1815375308907786402</li>
        <li>Last Price Increase: 2024-09-03</li>
    </ul>
</details>

**Comment by AndrewGable** [comment]
Does this only happen in HybridApp, or standalone NewDot?

**Comment by danielrvidal** [comment]
It's only happening in Hybrid app. I just opened the app and it's still happening to me so let me know if I can do anything to help diagnose. 
![IMG_6262](https://github.com/user-attachments/assets/3f138f52-cff3-4fa3-af41-228ce0a0e8e0)

**Comment by AndrewGable** [comment]
cc @staszekscp @mateuuszzzzz - Can one of you check this?

**Comment by AndrewGable** [comment]
I actually had this happen to me on Web, so I don't think this is limited to HybridApp and I think we should be looking at this in a larger context.

e.g. It says receipt scanning in LHN and when I click onto the report. 

Only once I click into the report, does it "refresh" and show that the SmartScan has finished.

![Google Chrome 2024-07-26 at 08 30 35](https://github.com/user-attachments/assets/a40489c7-eaa8-430b-b6a5-fe18d2c2b124)

![Google Chrome 2024-07-26 at 08 30 40](https://github.com/user-attachments/assets/0d9737e8-ebbb-4c48-b614-2a1579c68224)

**Comment by techievivek** [comment]
I have removed Hybrid web from the title.

**Comment by techievivek** [comment]
Not overdue, still waiting for the proposals on this one.

**Comment by techievivek** [comment]
Seems like a backend issue, I will take a look myself.

**Comment by techievivek** [comment]
Not overdue, couldn't reproduce this earlier. I will give it another look this week.

**Comment by techievivek** [comment]
Not overdue, still needs to reproduce.

**Comment by trjExpensify** [comment]
#newdot-quality autoassigner on this, moving it over.  CC: @muttmuure for where you want to put it.

**Comment by techievivek** [comment]
I am still unable to reproduce this on my end. 😞 I tried on an existing chat report with scanned receipts from an earlier data and on a new chat report both locally and on staging.

<img width="694" alt="Screenshot 2024-08-09 at 1 05 37 PM" src="https://github.com/user-attachments/assets/fdf9f6bc-56f4-4ff0-a70f-c1d1cb3c85e7">

<img width="1054" alt="Screenshot 2024-08-09 at 1 05 31 PM" src="https://github.com/user-attachments/assets/5d591d46-3c85-46a2-8158-af2fde57ebbe">


<img width="938" alt="Screenshot 2024-08-09 at 1 07 03 PM" src="https://github.com/user-attachments/assets/c627a6c2-da3e-4f61-8922-d1008f7816bf">

**Comment by techievivek** [comment]
Is this happening specifically on workspace chats? Let me give it a try there.

**Comment by techievivek** [comment]
No luck even with workspace chat. Am I holding anything wrong? Or do we have an applause account that I can use to test this?
<img width="777" alt="Screenshot 2024-08-09 at 2 06 44 PM" src="https://github.com/user-attachments/assets/29c611aa-bd87-483f-9a6e-71c9b3adc948">
<img width="698" alt="Screenshot 2024-08-09 at 2 06 39 PM" src="https://github.com/user-attachments/assets/ea275e8f-bd8b-49c7-9eab-f0155568cdcf">

**Comment by muttmuure** [comment]
@danielrvidal can you share your onyx data in the Slack thread so we can triage this remotely please?

**Comment by danielrvidal** [comment]
Dropped the recent information in the thread. If you need me please ping me in Slack or via DM.

**Comment by danielrvidal** [comment]
Juist as an fyi, I dropped fresh logs in the slack thread so I think the team is still working through it.

**Comment by techievivek** [comment]
I tried reproducing this issue again today but had no luck. From the thread, it appears this is only happening on the hybrid app, but we don't have a solid root cause yet. Eduardo is already looking into it, so I'll ask in the thread if we can assign the GH issue to him.

**Comment by gedu** [comment]
Hey, I'm looking more into the ReconnectApp filling the network queue. This ReconnectApp can affect many different features, given that all the ReconnectApp request must be completed before handling new ones.
On the other hand, not sure if this issue is specific to the hybrid app, and given I don't have access and knowledge to it, probably would be better if someone else can take it

**Comment by slafortune** [comment]
Closing this in favor of these two separate issues created after much discussion here - https://expensify.slack.com/archives/C05LX9D6E07/p1721152209600379 - 
https://github.com/Expensify/App/issues/47740
https://github.com/Expensify/App/issues/47742

**Comment by muttmuure** [comment]
Let's reopen this

**Comment by muttmuure** [comment]
@rinej can you comment on this so I can assign you?

**Comment by rinej** [comment]
Hello @muttmuure, sure please assign me. I've worked on tha logjam issue, so it might be related. 
Also please remember I will be OOO from tomorrow till the end of the week

**Comment by techievivek** [comment]
Not overdue, @rinej will start to work on it this week.

**Comment by rinej** [comment]
Based on the slack discussion, we added more logs in the retry logic to identify the root cause -> https://github.com/Expensify/App/pull/48983

**Comment by techievivek** [comment]
@rinej The PR is deployed to PROD. Did you get a chance to dig into this further, given that we now have better logs for it?

**Comment by rinej** [comment]
Right now we have to wait for the issue to happen again so we can dig into logs to confirm our investigation. 
Here is the slack message created by Matt -> https://callstack-hq.slack.com/archives/C05LX9D6E07/p1726224795425099

Also there is outgoing PR, which might also fix the issue  -> https://github.com/Expensify/App/pull/47913

**Comment by muttmuure** [comment]
This is the discussion [thread](https://github.com/Expensify/App/pull/47913) for this issue. The queue is not getting "stuck", the requests are getting stuck behind `ReconnectApp` calls. 

Putting this issue on HOLD

**Comment by mountiny** [comment]
This should be resolved as the reconnectApp calls been deduped, I will close this issue and feel free to reopen if this is still happening
